LabApiUtilities/src/main/com/microsoft/identity/labapi/utilities/client/LabClient.java

```diff
@@ -0,0 +1,188 @@
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.labapi.utilities.client;
+
+import com.microsoft.identity.internal.test.labapi.ApiException;
+import com.microsoft.identity.internal.test.labapi.Configuration;
+import com.microsoft.identity.internal.test.labapi.api.ConfigApi;
+import com.microsoft.identity.internal.test.labapi.api.CreateTempUserApi;
+import com.microsoft.identity.internal.test.labapi.api.LabSecretApi;
+import com.microsoft.identity.internal.test.labapi.model.ConfigInfo;
+import com.microsoft.identity.internal.test.labapi.model.SecretResponse;
+import com.microsoft.identity.internal.test.labapi.model.TempUser;
+import com.microsoft.identity.labapi.utilities.authentication.LabApiAuthenticationClient;
+import com.microsoft.identity.labapi.utilities.constants.TempUserType;
+import com.microsoft.identity.labapi.utilities.constants.UserType;
+import com.microsoft.identity.labapi.utilities.exception.LabApiException;
+import com.microsoft.identity.labapi.utilities.exception.LabError;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.NonNull;
+
+@AllArgsConstructor(access = AccessLevel.PUBLIC)
+public class LabClient implements ILabClient {
+
+    private final LabApiAuthenticationClient mLabApiAuthenticationClient;
+
+    /**
+     * Temp users API provided by Lab team can often take more than 10 seconds to return...hence, we
+     * are overriding the read timeout.
+     */
+    private static final int TEMP_USER_API_READ_TIMEOUT = (int) TimeUnit.SECONDS.toMillis(15);
+
+    @Override
+    public LabAccount getLabAccount(@NonNull final LabQuery labQuery) throws LabApiException {
+        final List<ConfigInfo> configInfos = fetchConfigsFromLab(labQuery);
+        // for each query, lab actually returns a list of accounts..all of which fit the criteria..
+        // usually we only need one such account, and hence over here we are just picking the first
+        // element of the list.
+        final ConfigInfo configInfo = configInfos.get(0);
+        return createAccount(configInfo);
```

Comment:
In this case you getting an account from the lab and populating a LabAccount DTO with that data.  I think the name here could be a bit confusing to causal readers....perhaps getLabAccountObject?  I think this ia nit overall...

Reply:
Addressed here: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1406/commits/ad75124d3f31efb65375ebad81874cea4eb55d1e